### ISSUE TITLE
Update from 0.18.0 to 0.18.3 break global Promise object 

### ISSUE THREAD

**Issue by super-ienien** [dashersw/cote#147]
Hi,

I can't update cote because the last version (or a version between 0.18.0 and 0.18.3) is replacing the global Promise object.

I think a library should not interfer with global space.  I don't know why the global Promise object is replaced, may be it's the es7 polyfills ?

In my case i'm replacing the native Promise with Bluebird at the very begining of my program. but when i'm loading cote the global Promise object is replaced by another Promise object wich is neither the native Promise or Bluebird.

Thanks

**Comment by dashersw** [comment]
Which version of node are you using? I've come up with the following example which hopefully illustrates your situation, but it doesn't override bluebird Promise.

```js
global.Promise = require("bluebird")

const cote = require('cote')

const req = new cote.Requester({ name: 'requester' })
```

**Comment by dashersw** [comment]
If you have an example repo, that also would help tremendously.

**Comment by dashersw** [comment]
Also, please note that cote doesn't support Node.js versions lower than 6.

**Comment by super-ienien** [comment]
I am running it on node v11.4.0

[i made a repo](https://github.com/super-ienien/cote-issue147)

Thanks a lot

**Comment by dashersw** [comment]
Thank you, I was able to reproduce the problem with this. I'll fix it over the weekend.

**Comment by dashersw** [comment]
@super-ienien I've released cote 0.18.4 that's supposed to fix this issue. Could you test?

**Comment by super-ienien** [comment]
Thanks a lot, the problem is gone with cote 0.18.4 ;)
